tests/pyfunc/test_spark.py

```diff
@@ -262,6 +262,40 @@ def predict(self, context, model_input):
         assert result.res[0] == "a,b"
 
 
+def test_spark_udf_with_struct_return_type(spark):
+    class TestModel(PythonModel):
+        def predict(self, context, model_input):
+            input_len = len(model_input)
+            return {
+                "r1": [1] * input_len,
+                "r2": [1.5] * input_len,
+                "r3": [[1, 2]] * input_len,
+                "r4": [np.array([1.5, 2.5])] * input_len,
+            }
+
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model("model", python_model=TestModel())
+
+        udf = mlflow.pyfunc.spark_udf(
+            spark,
+            "runs:/{}/model".format(run.info.run_id),
+            result_type="r1 int, r2 float, r3 array<long>, r4 array<double>",
```

Comment:
Does the user always have to pass this kind of "<key> <type>" specification for Spark StructType, or can we support letting the user pass the string "StructType"? It would be nice if users didn't have to specify every key & type.

Seems like this should be possible since we can read the result dictionary here: https://github.com/mlflow/mlflow/pull/7527/files#diff-592813f9fb198dda0982794f2221a408f316709190b3bf02c0e77eb2c1107075R1057-R1058

Reply:
Inferring field type is possible,

But if we support inferring struct field type, we should also support inferring primitive type / array type, why not do it in follow-up task (no need go into 2.1) so that these cases they can share some code ?

Reply:
How do we infer the field type?

Reply:
e.g., if the field values is a numpy array, we can check the numpy array dtype,
if it is a list, we can check first element type, or coalesce types from all elements in the list,
it needs to add a tons of code to handle this.

Reply:
Got it. Okay, makes sense. Let's do that later. Can we update the docs for `spark_udf` to include a more specific example of how to pass `result_type` for struct type? E.g. something like the example in your test case?